bibliopixel/layout/geometry/matrix.py

```diff
@@ -101,30 +98,28 @@ def combine_cols(cols):
 
     if isinstance(config, dict):
         matrix_result = make_matrix_coord_map(**config)
+
     else:
         for row in config:
             if isinstance(row, dict):  # is configs
                 row = populate_config(row)
-                matrix_result, matrix_offset = add_row(matrix_offset, matrix_result, make_matrix_coord_map(**row))
+                matrix_offset = add_row(
+                    matrix_offset, matrix_result, make_matrix_coord_map(**row))
+
             elif isinstance(row, list):  # is row
                 cols = []
                 for config in row:
                     config = populate_config(config)
                     cols.append(make_matrix_coord_map(**config))
-                matrix_result, matrix_offset = add_row(matrix_offset, matrix_result, combine_cols(cols))
 
-    matrix_result = rotate_and_flip(matrix_result, rotation, y_flip)
+                matrix_offset = add_row(
```

Comment:
you aren't capturing matrix_result from add_row

Reply:
I don't need to - it never changes.